3D Chess/Assets/Scripts/Piece.cs

```diff
@@ -27,4 +28,25 @@ private void Awake()
     {
         events = FindObjectOfType<Events>();
     }
+
+    private void Start()
+    {
+        Entity.PropertyChanged += OnPropertyChanged;
+    }
+
+    private void OnPropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        switch (e.PropertyName)
+        {
+            case nameof(Entity.IsValid):
+                if (!Entity.IsValid)
+                {
+                    Destroy(gameObject);
+                }
+                break;
```

Comment:
Might be worth adding a comment to mention this case handles the object getting deleted.

Reply:
I have added this and many other comments, alongside some clean up, with 7b88476.